server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -160,8 +161,7 @@ public TSOpenSessionResp openSession(TSOpenSessionReq req) throws TException {
       zoneIds.set(config.getZoneID());
       initForOneSession();
     } else {
-      tsStatus = new TS_Status(TS_StatusCode.ERROR_STATUS);
-      tsStatus.setErrorMessage("login failed. Username or password is wrong.");
+      tsStatus = getErrorStatus(ErrorCode.WRONG_LOGIN_PASSWORD_ERROR, "login failed. Username or password is wrong.");
```

Comment:
A more thorough solution: make ErrorCode (or StatusCode more precisely) an enum and bind the message with each code.

Reply:
I didn't bind the message because I thought some error code may have more equivalent messages. For example, `FETCH_METADATA_ERROR` may have messages `Unsupported fetch metadata operation ...` or `Failed to fetch metadata because...`. If the error codes and messages are bound, then maybe the number of error codes will be larger and more detailed.

Reply:
Yes, I think the more clear the codes, the better, which will help the user faster locate the problem.